src/metadataGeneration/customAttribute.ts

```diff
@@ -0,0 +1,67 @@
+import * as ts from 'typescript';
+import { getInitializerValue } from './initializer-value';
+import { Tsoa } from './tsoa';
+
+const getArgumentValue = (argument: ts.Expression) => {
+  switch (argument.kind as ts.SyntaxKind) {
+    case ts.SyntaxKind.StringLiteral: {
+      const argumentValue = (argument as any).text;
+
+      return argumentValue;
+    }
+
+    case ts.SyntaxKind.ObjectLiteralExpression: {
```

Comment:
This block contains a big amount of assumptions disguised by the `any` cast.
Property is `ObjectLiteralElementLike` which can be `PropertyAssignment | ShorthandPropertyAssignment | SpreadAssignment | MethodDeclaration | AccessorDeclaration`

This piece of code is only handling PropertyAssignment, so it should throw on any thing else.